inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManager.java

```diff
@@ -0,0 +1,495 @@
+
+    private final PeriodicTask updateMetaDataWorker;
+    private volatile List<String> toBeSelectFetchers = new ArrayList<>();
+    private boolean stopAssign = false;
+
+    public InlongSingleTopicManager(ClientContext context, QueryConsumeConfig queryConsumeConfig) {
+        super(context, queryConsumeConfig);
+        updateMetaDataWorker = new UpdateMetaDataThread(context.getConfig().getUpdateMetaDataIntervalSec(),
+                TimeUnit.SECONDS);
+        String threadName = "sortsdk_inlongtopic_manager_" + context.getConfig().getSortTaskId()
+                + "_" + StringUtil.formatDate(new Date(), "yyyy-MM-dd HH:mm:ss");
+        updateMetaDataWorker.start(threadName);
+    }
+
+    private void updateToBeSelectFetchers(Collection<String> c) {
+        toBeSelectFetchers = new ArrayList<>(c);
+    }
+
+    private boolean initFetcher(InLongTopicFetcher fetcher, InLongTopic inLongTopic) {
+        if (InlongTopicTypeEnum.PULSAR.getName().equalsIgnoreCase(inLongTopic.getTopicType())) {
+            LOGGER.info("create fetcher topic is pulsar {}", inLongTopic);
+            return fetcher.init(pulsarClients.get(inLongTopic.getInLongCluster().getClusterId()));
+        } else if (InlongTopicTypeEnum.KAFKA.getName().equalsIgnoreCase(inLongTopic.getTopicType())) {
+            LOGGER.info("create fetcher topic is kafka {}", inLongTopic);
+            return fetcher.init(inLongTopic.getInLongCluster().getBootstraps());
+        } else if (InlongTopicTypeEnum.TUBE.getName().equalsIgnoreCase(inLongTopic.getTopicType())) {
+            LOGGER.info("create fetcher topic is tube {}", inLongTopic);
+            return fetcher.init(tubeFactories.get(inLongTopic.getInLongCluster().getClusterId()));
+        } else {
+            LOGGER.error("create fetcher topic type not support " + inLongTopic.getTopicType());
+            return false;
+        }
+    }
+
+    @Override
+    public InLongTopicFetcher addFetcher(InLongTopic inLongTopic) {
+
+        try {
+            InLongTopicFetcher result = fetchers.get(inLongTopic.getTopicKey());
+            if (result == null) {
+                // create fetcher (pulsar,tube,kafka)
+                InLongTopicFetcher inLongTopicFetcher = createInLongTopicFetcher(inLongTopic);
+                InLongTopicFetcher preValue = fetchers.putIfAbsent(inLongTopic.getTopicKey(), inLongTopicFetcher);
+                LOGGER.info("addFetcher :{}", inLongTopic.getTopicKey());
+                if (preValue != null) {
+                    result = preValue;
+                    if (inLongTopicFetcher != null) {
+                        inLongTopicFetcher.close();
+                    }
+                    LOGGER.info("addFetcher create same fetcher {}", inLongTopic);
+                } else {
+                    result = inLongTopicFetcher;
+                    if (result != null
+                            && !initFetcher(result, inLongTopic)) {
```

Comment:
Those two lines can be merged into one line.

Reply:
fixed, thx